File: lab2zad2.cpp
#include <iostream>
using namespace std;

class Tablica {
private:
    int* tablica;
    int rozmiar;

public:
    Tablica() : rozmiar(10) {
        tablica = new int[rozmiar];
        cout << "Tablica o rozmiarze 10 została utworzona." << endl;
    }

    Tablica(int rozmiar) : rozmiar(rozmiar) {
        tablica = new int[rozmiar];
        cout << "Tablica o rozmiarze " << rozmiar << " została utworzona." << endl;
    }

    ~Tablica() {
        delete[] tablica;
        cout << "Pamięć została zwolniona." << endl;
    }

    void setElement(int index, int value) {
        if (index >= 0 && index < rozmiar) {
            tablica[index] = value;
        }
    }

    void display() const {
        for (int i = 0; i < rozmiar; ++i) {
            cout << tablica[i] << " ";
        }
        cout << endl;
    }
};

int main() {
    Tablica tab1;
    tab1.setElement(0, 10);
    tab1.display();

    Tablica tab2(5);
    tab2.setElement(2, 20);
    tab2.display();

    return 0;
}

File: lab2zad2.java
public class Tablica {
    private int[] tablica;

    public Tablica() {
        tablica = new int[10];
        System.out.println("Tablica o rozmiarze 10 została utworzona.");
    }

    public Tablica(int rozmiar) {
        tablica = new int[rozmiar];
        System.out.println("Tablica o rozmiarze " + rozmiar + " została utworzona.");
    }

    public void setElement(int index, int value) {
        if (index >= 0 && index < tablica.length) {
            tablica[index] = value;
        }
    }

    public void display() {
        for (int i : tablica) {
            System.out.print(i + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        Tablica tab1 = new Tablica();
        tab1.setElement(0, 10);
        tab1.display();

        Tablica tab2 = new Tablica(5);
        tab2.setElement(2, 20);
        tab2.display();
    }
}
